src/main/java/org/elasticsearch/cluster/metadata/MetaData.java

```diff
@@ -674,6 +674,15 @@ public String resolveIndexRouting(@Nullable String routing, String aliasOrIndex)
 
             aliasesOrIndices = convertFromWildcards(aliasesOrIndices, indicesOptions);
         }
+
+        if (aliasesOrIndices == null || aliasesOrIndices.length == 0) {
+            if (!indicesOptions.allowNoIndices()) {
+                throw new ElasticsearchIllegalArgumentException("No indices were specified and wildcard expansion is disabled.");
```

Comment:
s/No/no we tend to have all lowercases logs/exceptions. Not that we are always consistent with this, but that is the preference.